### ISSUE TITLE
Inconsistent block starting line behaviour for classes

### ISSUE THREAD

**Issue by cwiede** [pylint-dev/pylint#10199]
### Bug description

```python
"""
docstring
"""

class FOO: # pylint: disable=invalid-name
    """
    docstring
    """

    def camelCase(self):
        "docstring"
        return

    def good_function(self):
        "docstring"
        return

class BAR(FOO): # pylint: disable=invalid-name
    """
    docstring
    """

    def camelCase2(self):
        "docstring"
        return

    def good_function2(self):
        "docstring"
        return
```

### Configuration

```ini

```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
************* Module test_pylint
test_pylint.py:23:4: C0103: Method name "camelCase2" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

It is surprising for me that the invalid name is only outputted for class BAR and not for class FOO. From the documentation I read

> If the violation occurs on a block starting line, then it applies only to that line
 
and so I would expect that the disable=invalid-name statements only apply to the single lines where the classes are defined and not to the whole block of the class.

### Pylint version

```shell
pylint 3.3.3
astroid 3.3.8
Python 3.13.1 (main, Jan 14 2025, 22:47:35) [MSC v.1942 64 bit (AMD64)]
```

### Additional dependencies

```python
astroid==3.3.8
colorama==0.4.6
dill==0.3.9
isort==5.13.2
mccabe==0.7.0
platformdirs==4.3.6
pylint==3.3.3
tomlkit==0.13.2
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. 

The [message control doc](https://github.com/pylint-dev/pylint/blob/main/doc/user_guide/messages/message_control.rst) doesn't really talk about scope for classes which is why I added a need specification label so we define how to suppress a message for a whole class. 

I thought it was:
```python
class Fruit:
    # pylint: disable=my-message
    ...
```

But it seems that someone thought that this made sense and implemented it (or maybe the underlying ast implicitly makes it work that way, I didn't look in detail):
```python
class Fruit: # pylint: disable=my-message
    ...
```

In any case there's an inconsistency right now for class vs inherited class.
